### PR TITLE
Added getters for configs, labels for distribution metric.

### PR DESCRIPTION
**Pull request by apoorvmittal10**
- Added `getters` for `ingestion spec` configs.
- Added `labels` map for `distribution` metric.

- Added addition `test cases`.

### CODE REVIEW

File: extensions-contrib/opencensus-extensions/src/test/java/org/apache/druid/data/input/opencensus/protobuf/OpenCensusProtobufInputRowParserTest.java [hunk 1]
```diff
@@ -124,6 +125,32 @@ public void testGaugeParse() throws Exception
...
+    ByteArrayOutputStream out = new ByteArrayOutputStream();
+    metric.writeTo(out);
+
+    InputRow row = parser.parseBatch(ByteBuffer.wrap(out.toByteArray())).get(0);
+    Assert.assertEquals(dateTime.getMillis(), row.getTimestampFromEpoch());
+
+    assertDimensionEquals(row, "name", "metric_gauge_int64");
+    assertDimensionEquals(row, "foo_key", "foo_value");
+
+    Assert.assertEquals(1000, row.getMetric("value").intValue(), 0.0);
```
Comment (xvrl) [inline extensions-contrib/opencensus-extensions/src/test/java/org/apache/druid/data/input/opencensus/protobuf/OpenCensusProtobufInputRowParserTest.java:151]: for integer comparisons, the delta (last argument to assertEquals) is unnecessary

File: extensions-contrib/opencensus-extensions/src/test/java/org/apache/druid/data/input/opencensus/protobuf/OpenCensusProtobufInputRowParserTest.java [hunk 2]
```diff
@@ -124,6 +125,32 @@ public void testGaugeParse() throws Exception
...
+  {
+    //configure parser with desc file
+    OpenCensusProtobufInputRowParser parser = new OpenCensusProtobufInputRowParser(parseSpec, null, null, "");
+
+    DateTime dateTime = new DateTime(2019, 07, 12, 9, 30, ISOChronology.getInstanceUTC());
+
+    Timestamp timestamp = Timestamp.newBuilder().setSeconds(dateTime.getMillis() / 1000)
+        .setNanos((int) ((dateTime.getMillis() % 1000) * 1000000)).build();
+
+    System.out.println(timestamp.getSeconds() * 1000);
```
Comment (xvrl) [inline extensions-contrib/opencensus-extensions/src/test/java/org/apache/druid/data/input/opencensus/protobuf/OpenCensusProtobufInputRowParserTest.java:139]: please remove any `System.out` statements from test code before committing. If you need to output anything use a logger. In this case, I don't think there is much value in these outputs.

File: extensions-contrib/opencensus-extensions/src/test/java/org/apache/druid/data/input/opencensus/protobuf/OpenCensusProtobufInputRowParserTest.java [hunk 3]
```diff
@@ -154,7 +181,38 @@ public void testSummaryParse() throws Exception
...
+
+    List<InputRow> rows = parser.parseBatch(ByteBuffer.wrap(out.toByteArray()));
+
+    Assert.assertEquals(2, rows.size());
+
+    InputRow row = rows.get(0);
+    Assert.assertEquals(dateTime.getMillis(), row.getTimestampFromEpoch());
+    assertDimensionEquals(row, "name", "metric_distribution-count");
+    assertDimensionEquals(row, "foo_key", "foo_value");
+    Assert.assertEquals(100, row.getMetric("value").intValue(), 0.0);
```
Comment (xvrl) [inline extensions-contrib/opencensus-extensions/src/test/java/org/apache/druid/data/input/opencensus/protobuf/OpenCensusProtobufInputRowParserTest.java:209]: same thing about removing deltas when comparing integer values

File: extensions-contrib/opencensus-extensions/src/test/java/org/apache/druid/data/input/opencensus/protobuf/OpenCensusProtobufInputRowParserTest.java (hunk 2 above)
Comment (xvrl) [reply extensions-contrib/opencensus-extensions/src/test/java/org/apache/druid/data/input/opencensus/protobuf/OpenCensusProtobufInputRowParserTest.java:139]: let's also remove that statement from `testDoubleGaugeParse` above.

File: extensions-contrib/opencensus-extensions/src/main/java/org/apache/druid/data/input/opencensus/protobuf/OpenCensusProtobufInputRowParser.java [hunk 4]
```diff
@@ -85,6 +85,24 @@ public ParseSpec getParseSpec()
     return parseSpec;
   }
 
+  @JsonProperty
+  public String getMetricDimension()
```
Comment (xvrl) [inline extensions-contrib/opencensus-extensions/src/main/java/org/apache/druid/data/input/opencensus/protobuf/OpenCensusProtobufInputRowParser.java:89]: can we add a test for serialization / de-serialization? Here's an example on how to test those things https://github.com/confluentinc/druid/blob/6163335cd4a8ab47b8c06c7bc47c9eabbee27985/extensions-contrib/confluent-extensions/src/test/java/io/confluent/druid/transform/ExtractTransformTest.java#L138-L155

File: extensions-contrib/opencensus-extensions/src/test/java/org/apache/druid/data/input/opencensus/protobuf/OpenCensusProtobufInputRowParserTest.java (hunk 2 above)
Comment (apoorvmittal10) [reply extensions-contrib/opencensus-extensions/src/test/java/org/apache/druid/data/input/opencensus/protobuf/OpenCensusProtobufInputRowParserTest.java:139]: Yeah, just didn't notice the statement in both methods.

File: extensions-contrib/opencensus-extensions/src/test/java/org/apache/druid/data/input/opencensus/protobuf/OpenCensusProtobufInputRowParserTest.java (hunk 1 above)
Comment (apoorvmittal10) [reply extensions-contrib/opencensus-extensions/src/test/java/org/apache/druid/data/input/opencensus/protobuf/OpenCensusProtobufInputRowParserTest.java]: Done.

File: extensions-contrib/opencensus-extensions/src/main/java/org/apache/druid/data/input/opencensus/protobuf/OpenCensusProtobufInputRowParser.java (hunk 4 above)
Comment (apoorvmittal10) [reply extensions-contrib/opencensus-extensions/src/main/java/org/apache/druid/data/input/opencensus/protobuf/OpenCensusProtobufInputRowParser.java:90]: Done.

**Pull request merged**